Tribler/community/market/core/order.py

```diff
@@ -0,0 +1,384 @@
+        assert isinstance(order_number, OrderNumber), type(order_number)
+
+        self._trader_id = trader_id
+        self._order_number = order_number
+
+    @property
+    def trader_id(self):
+        """
+        :rtype: TraderId
+        """
+        return self._trader_id
+
+    @property
+    def order_number(self):
+        """
+        :rtype: OrderNumber
+        """
+        return self._order_number
+
+    def __str__(self):
+        """
+        format: <trader_id>.<order_number>
+        """
+        return "%s.%s" % (self._trader_id, self._order_number)
+
+    def __eq__(self, other):
+        if not isinstance(other, OrderId):
+            return NotImplemented
+        elif self is other:
+            return True
+        else:
+            return (self._trader_id, self._order_number) == \
+                   (other._trader_id, other._order_number)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash((self._trader_id, self._order_number))
+
+
+class Order(object):
+    """Class for representing an ask or a bid created by the user"""
+
+    def __init__(self, order_id, price, quantity, timeout, timestamp, is_ask):
+        """
+        :param order_id: An order id to identify the order
+        :param price: A price to indicate for which amount to sell or buy
+        :param quantity: A quantity to indicate how much to sell or buy
+        :param timeout: A timeout when this tick is going to expire
+        :param timestamp: A timestamp when the order was created
+        :param is_ask: A bool to indicate if this order is an ask
+        :type order_id: OrderId
+        :type price: Price
+        :type quantity: Quantity
+        :type timeout: Timeout
+        :type timestamp: Timestamp
+        :type is_ask: bool
+        """
+        super(Order, self).__init__()
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        assert isinstance(order_id, OrderId), type(order_id)
+        assert isinstance(price, Price), type(price)
+        assert isinstance(quantity, Quantity), type(quantity)
+        assert isinstance(timeout, Timeout), type(timeout)
+        assert isinstance(timestamp, Timestamp), type(timestamp)
+        assert isinstance(is_ask, bool), type(is_ask)
+
+        self._order_id = order_id
+        self._price = price
+        self._quantity = quantity
+        self._reserved_quantity = Quantity(0, quantity.wallet_id)
+        self._traded_quantity = Quantity(0, quantity.wallet_id)
+        self._timeout = timeout
+        self._timestamp = timestamp
+        self._completed_timestamp = None
+        self._is_ask = is_ask
+        self._reserved_ticks = {}
+        self._cancelled = False
+
+    @classmethod
+    def from_database(cls, data, reserved_ticks):
+        """
+        Create an Order object based on information in the database.
+        """
+        order_id = OrderId(TraderId(str(data[0])), OrderNumber(data[1]))
```

Comment:
Could you unpack the data list into named variables? This makes the code infinitely easier to check for errors.